tests/integration/error_handler_test.ts

```diff
@@ -0,0 +1,127 @@
+import { Rhum, TestHelpers } from "../deps.ts";
+import { ErrorHandler, Errors, Response, Server } from "../../mod.ts";
+
+class MyErrorHandler extends ErrorHandler {
+  public catch(error: Error, _request: Request, response: Response) {
+    if (error instanceof Errors.HttpError) {
+      response.status = error.code;
+    }
+    response.json({ error: error.message });
+  }
+}
```

Comment:
why error is Error here and is HttpError in default ErrorHandler ?
if is simple Error thrown, this ErrorHandler return 200 ?

Reply:
I think use Error everywhere and set status to 500 if not an HttpError, see my last commits, it's better for me.

Reply:
So there's two approaches here:
1. Catch on every error. This means the server will never stop working, regardless of the error. If an error is thrown just send `500`.
2. Catch only HttpError. This can make the server crash if, for instance, a database errors and the developer doesn't worry about surrounding external calls with `try...catch`.

My opinion is that option 1 would be the most interesting. Ideally the server never goes down. If an error occurs that was not expected, and `Internal Server Error Exception` shall be sent to the user.

Reply:
Same opinion, that's exactly what I did.